### PR TITLE
Integrate `cargo-xwin` for cross compiling to Windows MSVC targets

### LINKED ISSUE #830: Better support for cross compiling to Windows
1. Cross compile to MSVC targets can be done with integrating [cargo-xwin](https://github.com/messense/cargo-xwin).
2. Cross compile to GNU targets can already be done with `--zig` I think.
3. ~~Even with abi3 an interpreter is required on Windows, maybe we can use something like the [python3-dll-a](https://github.com/ravenexp/python3-dll-a) project to generate an Windows import library for `python3.dll` automatically.~~ Resolved in https://github.com/PyO3/pyo3/pull/2282

### PR DESCRIPTION
**Pull request by messense**
Closes #830

### CODE REVIEW

**Comment by messense** [comment]
cc @ravenexp

**Comment by messense** [comment]
I think `zig dlltool` is same as `llvm-dlltool` and they are not compatible with mingw64's dlltool, thus use `zig dlltool` for windows gnu targets isn't possible? @ravenexp

**Comment by ravenexp** [comment]
It depends on what linker `zig cc` uses for its MinGW targets. I think Zig always links with `lld`, which uses the Windows import library format, same as `LINK.EXE`. MinGW-format import libraries are only suitable when `x86_64-w64-mingw32-gcc` is used as the linker.

**Comment by ravenexp** [comment]
> It depends on what linker `zig cc` uses for its MinGW targets. I think Zig always links with `lld`, which uses the Windows import library format, same as `LINK.EXE`.

I checked this directly just to be sure, and `zig cc` indeed uses `lld-link` when compiling for `-target x86_64-windows-gnu`:

```
$ zig cc -v -target x86_64-windows-gnu test.c

--skip--
lld-link -ERRORLIMIT:0 -NOLOGO -STACK:16777216 -MACHINE:X64 -OUT:/x/.cache/zig/o/d5879460d6961e986517a114b13aaef2/a.exe -IMPLIB:/x/.cache/zig/o/d5879460d6961e986517a114b13aaef2/test.lib`
```

This means `dlltool` from MinGW binutils _can not_ be used with `cargo zigbuild` and `maturin --zig`.

File: src/compile.rs [hunk 1]
```diff
@@ -192,26 +192,50 @@ fn compile_target(
...
+
+    if context.zig {
+        // Pass zig command to downstream, eg. python3-dll-a
+        if let Ok((zig_cmd, zig_args)) = cargo_zigbuild::Zig::find_zig() {
+            let zig_cmd = if zig_args.is_empty() {
+                zig_cmd
+            } else {
+                format!("{} {}", zig_cmd, zig_args.join(" "))
+            };
+            build_command.env("ZIG_COMMAND", zig_cmd);
```
Comment (messense) [inline src/compile.rs:235]: @ravenexp `ZIG_COMMAND` is exposed here, it will be either `zig` or `python3 -m ziglang`.

Comment (ravenexp) [reply src/compile.rs:235]: Good, I'll implement support for `ZIG_COMMAND` in https://github.com/PyO3/python3-dll-a/pull/18.

**Pull request merged**